Je veux afficher un nbre random dans le père et un dans le fils

amostin opened the pull request:
Je compte prendre le [code openclassrooms](https://openclassrooms.com/fr/courses/1513891-la-programmation-systeme-en-c-sous-unix/1514339-les-processus#/id/r-1515141) pour le fork (car on lance proprement le process fils et pere avec des nom bien lisible. Puis je tape juste un generateur de nombre dedans

amostin committed:
Je veux afficher un nbre radom dans le père et un dans le fils

amostin committed:
Affiche un nbre random dans le fils

mais ça affiche  38Secteur 1: 38
au lieu de Secteur 1: 38

amostin commented:
### Voici l'output quand le fils contient la fct random et lappelle
![forkWTF](https://user-images.githubusercontent.com/40889985/61178243-319f5780-a5e9-11e9-99db-1bdf0ec72512.PNG)

amostin committed:
affiche 1 nombre par process

La fonction qui genere le random est en dehors des processus mais ils arrivent à l'appeler. Par contre c'est les memse nombres

amostin commented:
### Voici l'output quand le pere et le fils generent un nbre en utilisant la fonction qui est en dehors des process
> Je précise car c'est pe être la raison qui fait qu'on a que les même nombre
![forkWTF2](https://user-images.githubusercontent.com/40889985/61178281-898a8e00-a5ea-11e9-9f05-2cf2d9388eb3.PNG)

ATHOOS commented:
Pour ce qui est d'afficher qu'une seule fois le nombre, c'etait juste un printf qui est resté dans la fonction genere_sec_entre_min_max
Mais pour ce qui est du nombre identique entre le fils et le père, je me demande si c'est pas parce que il exécute le fils et le père en même temps, auquel cas, la fonction genere_sec_entre_min_max doit être appelée qu'une fois, et elle renvoie le même résultat aux deux.
Je vais essayer de creuser un peu la dessus

ATHOOS commented:
J'ai trouvé plusieurs fois ce genre de problème, mais rien n'est vraiment parfait, par exemple, on parle d'associer le pid du processus à srand() afin d'avoir une seed differente pour chacun des processus, mais le problème sera qu'on aura toujours le même nombre dans le processus. J'ai lu aussi qu'on pouvait executer un refresh toutes les nano-secondes, comme ca le programme arrive pas à utiliser deux fois la même seed pour srand. Mais bon faut réussir a faire ca, et je sais pas trop par où passer.

amostin commented:
### Ok ouais j'avis vu cette histoire de pid et ça me parait bien mais un peu confus quand meme
## Moi j'avais pensé a modifier la valeur du nbre random dans le fils et garder celui d'origine pour le pere

ATHOOS commented:
En soit, le père (dans la version finale du projet) n'aura pas besoin d'acceder à la fonction qui genere les temps, puisqu'il ne fera que afficher les temps renvoyés par les fils. 
Donc ouais je pense qu'il faudait qu'on utilise la seed du pid pour chaque fils, et qu'on trouve le moyen de refresh pour chaque fils 
J'ai peut être une idee de comment faire mais va falloir modifier le fonctionnement de la fonction genere_sec_entre_min_max

ATHOOS commented:
Edit : j'ai reussi a avoir des temps differents pour chacun des process, petit probleme, c'est que j'ai une valeur superieure a ce que je suis censé atteindre genre j'ai 41 secondes alors que je demande un temps compris en 35 et 40

ATHOOS commented:
@amostin faut que je te demande de m'expliquer le fonctionnement de ta fonction de generation de temps, parce que je comprend pas d'ou viens le problème
Plus particulierement cette ligne : 
`int sec = rand() % (max + 2 - min) + min;`

amostin commented:
Si on a rand() %10 on aura un nbre entre 0 et 10 du coup on prend ce nombre et on ajoute le min.
# je sais pas pourquoi ca va au dessus du max mais cest jamais plus que 1 au dessus donc alz non ?

ATHOOS committed:
nombres aleatoires gérés

fork fonctionnel, appel de la fonction de generation fonctionnel instanciée sur chaque extremite du fork

amostin committed:
ajout de commentaire pour srand() et min max

amostin commented:
close #36
